SamplesDashboard/SamplesDashboard/Services/AzureSdkService.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml;
+using System.Xml.Linq;
+
+namespace SamplesDashboard.Services
+{
+    public class AzureSdkService
+    {
+        private readonly IHttpClientFactory _clientFactory;
+        public AzureSdkService(IHttpClientFactory clientFactory)
+        {
+            _clientFactory = clientFactory;
+            
+        }
+        public async Task<Dictionary<string, string>> FetchAzureSdkVersions()
+        {
+            var httpClient = _clientFactory.CreateClient();
+
+            HttpResponseMessage responseMessage = await httpClient.GetAsync("https://raw.githubusercontent.com/Azure/azure-sdk-for-net/master/eng/Packages.Data.props");
+
+            if (!responseMessage.IsSuccessStatusCode)
+            {
+                return null;
+            }
+            string fileContents = await responseMessage.Content.ReadAsStringAsync();
+            XmlDocument xml = new XmlDocument();
+            xml.LoadXml(fileContents);
+
+            XmlNodeList data = xml.GetElementsByTagName("PackageReference");
+            Dictionary<string, string> packages = new Dictionary<string, string>();
+
+            foreach (XmlNode node in data)
+            {
+                if (!packages.ContainsKey(node.Attributes["Update"].Value))
+                {
+                    packages.Add(node.Attributes["Update"].Value, node.Attributes["Version"].Value);
+                }
+            }
+
+            return packages;
+        }
+
+        /// <summary>
+        /// This method gets a dictionary of packages and sets the version used
+        /// </summary>
+        /// <param name="packageName"></param>
+        /// <returns>azure sdk version</returns>
+        public async Task<string> GetAzureSdkVersions(string packageName)
+        {
+            string azureSdkVersion = String.Empty;
+
+            var sdks = await FetchAzureSdkVersions();
```

Comment:
Every time you want to do a lookup, it seems that you are downloading the file again. This can be optimised by downloading the file once and caching it for the bunch of lookups to be performed.

Reply:
Okay